### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by FrankMugagga**
# Project requirements

- [ ] Remove current_user method in ApplicationController, devise will provide us with one.

- [ ] Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- [ ] Can register a new user.
- [ ] User logs in with a combination of email and password.
- [ ] Hashed password should be stored in the database.
- [ ] Ask for confirmation of email.
- [ ] Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (laiifuu) [inline app/models/user.rb]: As stated in the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#:~:text=Ask%20for%20confirmation%20of%20email.), an email of confirmation should be sent after a user registers. You should add `:confirmable` here.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (laiifuu) [inline app/views/devise/registrations/new.html.erb]: As requested in the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#:~:text=Modify%20the%20views%20in%20devise%20for%20registration%20and%20login%20to%20match%20the%20wireframes%20from%20the%20Sneak%20Peek%20and%20your%20styling.), you should style the view for registration to match the wireframe, currently it looks like this: 
image

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: 

## Hi team :wave:  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
- [x] Devise correctly installed :heavy_check_mark: 
- [x] Users can login and sign up to the application :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Email confirmation is enabled ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- From the comment you added you can see the confirmation message in you terminal when you click the email will be confirmed 👍🏿 
Screenshot_112


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**